R/RLearner_classif_mxff.R

```diff
@@ -0,0 +1,416 @@
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "conv.pad32", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.kernel11", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.kernel12", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.kernel21", lower = 1L,
+        requires = quote(conv.layer2 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.kernel22", lower = 1L,
+        requires = quote(conv.layer2 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.kernel31", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.kernel32", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.stride11", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.stride12", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.stride21", lower = 1L,
+        requires = quote(conv.layer2 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.stride22", lower = 1L,
+        requires = quote(conv.layer2 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.stride31", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.stride32", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.pad11", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.pad12", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.pad21", lower = 1L,
+        requires = quote(conv.layer2 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.pad22", lower = 1L,
+        requires = quote(conv.layer2 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.pad31", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeIntegerLearnerParam(id = "pool.pad32", lower = 1L,
+        requires = quote(conv.layer3 == TRUE)),
+      makeDiscreteLearnerParam(id = "pool.type1", default = "max",
+        values = c("max", "avg", "sum"),
+        requires = quote(conv.layer1 == TRUE)),
+      makeDiscreteLearnerParam(id = "pool.type2", default = "max",
+        values = c("max", "avg", "sum"),
+        requires = quote(conv.layer2 == TRUE)),
+      makeDiscreteLearnerParam(id = "pool.type3", default = "max",
+        values = c("max", "avg", "sum"),
+        requires = quote(conv.layer3 == TRUE)),
+      # other hyperparameters
+      makeNumericLearnerParam(id = "validation.set"),
+      makeIntegerLearnerParam(id = "early.stop.badsteps", lower = 1),
+      makeLogicalLearnerParam(id = "early.stop.maximize", default = TRUE),
+      makeNumericVectorLearnerParam(id = "dropout", lower = 0, upper = 1 - 1e-7),
```

Comment:
Hm, so dropout might have different lengths (depending on if I want separate dropout ratios in every layer). This is again a bit annoying to tune. 
Maybe have them as separate parameters per layer and have an additional dropout.global. The separate layers then `require(dropout.global == FALSE)`.